scapy/layers/dns.py

```diff
@@ -1259,12 +1259,36 @@ def is_request(self, req):
     def make_reply(self, req):
         if IPv6 in req:
             resp = IPv6(dst=req[IPv6].src, src=self.src_ip6)
-        else:
+        elif IP in req:
             resp = IP(dst=req[IP].src, src=self.src_ip)
-        resp /= UDP(sport=req.dport, dport=req.sport)
+        else:
+            warning("No IP or IPv6 layer in %s", req.command())
+            return
+        try:
+            resp /= UDP(sport=req[UDP].dport, dport=req[UDP].sport)
+        except IndexError:
+            warning("No UDP layer in %s", req.command(), exc_info=True)
+            return
         ans = []
-        req = req.getlayer(self.cls)
-        for rq in req.qd:
+        try:
+            req = req[self.cls]
+        except IndexError:
+            warning(
+                "No %s layer in %s",
+                self.cls.__name__,
+                req.command(),
+                exc_info=True,
+            )
+            return
+        try:
+            queries = req.qd
+        except AttributeError:
+            warning("No qd attribute in %s", req.command(), exc_info=True)
+            return
+        for rq in queries:
+            if isinstance(rq, Raw):
```

Comment:
Thanks!

I kind of started fixing a somewhat related issue (where malformed LLMNR queries/responses crash `tshark`) and in the process I thought it would be better to introduce something like `is_valid_dns_stuff` that could be used there and in `DNS_am.is_request` to filter out stuff like that before it reaches `DNS_am.make_reply`. This way `LLMNR.mysummary`, `DNS_am.is_request` and probably `DNS.mysummary` could rely on that function and reject malformed packets gracefully. It could be I'm missing something though.

Reply:
> I thought it would be better to introduce something like is_valid_dns_stuff

On second thought it doesn't seem to be a good idea after all. `DNS.mysummary` has to iterate over the lists anyway and something like https://github.com/secdev/scapy/pull/4012#discussion_r1283326882 should suffice there without any additional checks. `LLMNR.mysummary` can just inherit that.